runtime/sys.c

```diff
@@ -244,15 +251,22 @@ CAMLprim value caml_sys_file_exists(value name)
 #endif
   char_os * p;
   int ret;
+  value exn;
 
   if (! caml_string_is_c_safe(name)) return Val_false;
   p = caml_stat_strdup_to_os(String_val(name));
-  caml_enter_blocking_section();
+  exn = caml_enter_blocking_section_exn();
+  if (Is_exception_result(exn)) goto cleanup;
```

Comment:
Same remark as in the dynlink files: why changing the structure of the code?

```suggestion
  exn = caml_enter_blocking_section_exn();
  if (Is_exception_result(exn)) {
    caml_stat_free(p);
    caml_raise(Extract_exception(exn));
  }
```